### ISSUE TITLE
trace_attributes - edge_index is 18446744073709551615

### ISSUE THREAD

**Issue by M393** [valhalla/valhalla#3626]
<details>
<summary>trace_attributes request</summary>

```json
{
    "costing": "auto",
    "shape": [
        {
            "lat": 48.63998,
            "lon": 8.08265
        },
        {
            "lat": 48.63976,
            "lon": 8.08249
        },
        {
            "lat": 48.63962,
            "lon": 8.08252
        },
        {
            "lat": 48.63962,
            "lon": 8.08257
        },
        {
            "lat": 48.63954,
            "lon": 8.08277
        },
        {
            "lat": 48.63954,
            "lon": 8.0828
        },
        {
            "lat": 48.63947,
            "lon": 8.08299
        },
        {
            "lat": 48.63944,
            "lon": 8.08308
        }
    ]
}
```
</details>

At the fourth matched point the edge_index is `18446744073709551615`, which I guess means it is not available.
Is there a different meaning to when the edge_index is simply omitted?
I am assuming this value should not be returned.

<details>
<summary>trace_attributes response</summary>

```json
{
    "matched_points": [
        {
            "distance_from_trace_point": 1.433,
            "edge_index": 0,
            "type": "matched",
            "distance_along_edge": 0.323,
            "lat": 48.639975,
            "lon": 8.082668
        },
        {
            "distance_from_trace_point": 1.703,
            "edge_index": 0,
            "type": "matched",
            "distance_along_edge": 0.803,
            "lat": 48.639753,
            "lon": 8.082511
        },
        {
            "distance_from_trace_point": 2.379,
            "edge_index": 1,
            "type": "matched",
            "distance_along_edge": 0.791,
            "lat": 48.639601,
            "lon": 8.082505
        },
        {
            "distance_from_trace_point": 4.011,
            "edge_index": 18446744073709551615,
            "type": "interpolated",
            "distance_along_edge": 0.007,
            "lat": 48.639587,
            "lon": 8.082546
        },
        {
            "distance_from_trace_point": 2.521,
            "edge_index": 2,
            "type": "matched",
            "distance_along_edge": 0.299,
            "lat": 48.639520,
            "lon": 8.082755
        },
        {
            "distance_from_trace_point": 3.488,
            "edge_index": 2,
            "type": "interpolated",
            "distance_along_edge": 0.333,
            "lat": 48.639512,
            "lon": 8.082779
        },
        {
            "distance_from_trace_point": 2.668,
            "edge_index": 2,
            "type": "matched",
            "distance_along_edge": 0.606,
            "lat": 48.639448,
            "lon": 8.082974
        },
        {
            "distance_from_trace_point": 2.593,
            "edge_index": 2,
            "type": "matched",
            "distance_along_edge": 0.733,
            "lat": 48.639419,
            "lon": 8.083064
        }
    ],
    "edges": [
        {
            "end_node": {
                "transition_time": 0.541,
                "type": "street_intersection",
                "admin_index": 0,
                "elapsed_time": 5.558,
                "intersecting_edges": [
                    {
                        "road_class": "service_other",
                        "use": "path",
                        "begin_heading": 297,
                        "to_edge_name_consistency": false,
                        "from_edge_name_consistency": false,
                        "cyclability": "both",
                        "walkability": "both"
                    }
                ]
            },
            "names": [
                "Wiesenweg"
            ],
            "speed": 25,
            "road_class": "service_other",
            "begin_heading": 205,
            "begin_shape_index": 0,
            "traversability": "both",
            "use": "living_street",
            "drive_on_right": false,
            "surface": "paved_smooth",
            "density": 5,
            "sac_scale": 0,
            "length": 0.039,
            "lane_count": 1,
            "end_shape_index": 1,
            "shoulder": false,
            "bicycle_network": 0,
            "max_downward_grade": 32768,
            "end_heading": 205,
            "mean_elevation": 32768,
            "max_upward_grade": 32768,
            "travel_mode": "drive",
            "weighted_grade": 0.000,
            "way_id": 34598353,
            "id": 219989244290,
            "vehicle_type": "car"
        },
        {
            "end_node": {
                "transition_time": 0.037,
                "type": "street_intersection",
                "admin_index": 0,
                "elapsed_time": 8.116,
                "intersecting_edges": [
                    {
                        "road_class": "service_other",
                        "use": "living_street",
                        "begin_heading": 206,
                        "to_edge_name_consistency": true,
                        "from_edge_name_consistency": true,
                        "driveability": "both",
                        "cyclability": "both",
                        "walkability": "both"
                    }
                ]
            },
            "names": [
                "Wiesenweg"
            ],
            "speed": 25,
            "road_class": "service_other",
            "begin_heading": 140,
            "begin_shape_index": 1,
            "traversability": "both",
            "use": "living_street",
            "drive_on_right": false,
            "surface": "paved_smooth",
            "density": 5,
            "sac_scale": 0,
            "length": 0.014,
            "lane_count": 1,
            "end_shape_index": 3,
            "shoulder": false,
            "bicycle_network": 0,
            "max_downward_grade": 32768,
            "end_heading": 140,
            "mean_elevation": 32768,
            "max_upward_grade": 32768,
            "travel_mode": "drive",
            "weighted_grade": 0.000,
            "way_id": 34598353,
            "id": 3032857278850,
            "vehicle_type": "car"
        },
        {
            "end_node": {
                "transition_time": 0.000,
                "admin_index": 0,
                "elapsed_time": 14.274
            },
            "names": [
                "Wiesenweg"
            ],
            "speed": 25,
            "road_class": "service_other",
            "begin_heading": 116,
            "begin_shape_index": 3,
            "traversability": "both",
            "use": "living_street",
            "drive_on_right": false,
            "surface": "paved_smooth",
            "density": 5,
            "sac_scale": 0,
            "length": 0.043,
            "lane_count": 1,
            "end_shape_index": 4,
            "shoulder": false,
            "bicycle_network": 0,
            "max_downward_grade": 32768,
            "end_heading": 116,
            "mean_elevation": 32768,
            "max_upward_grade": 32768,
            "travel_mode": "drive",
            "weighted_grade": 0.000,
            "way_id": 34598353,
            "id": 218546403714,
            "vehicle_type": "car"
        }
    ],
    "admins": [
        {
            "state_code": "",
            "state_text": "None",
            "country_text": "None",
            "country_code": ""
        }
    ],
    "raw_score": 3.305,
    "shape": "m}vw{AwmilNnRxLdAn@lAkFrIu_@",
    "confidence_score": 1.000,
    "osm_changeset": 9233931469,
    "alternate_paths": []
}
```

**Comment by nilsnolde** [comment]
See here with more context: https://github.com/valhalla/valhalla/issues/3371

we have a few (minor) bugs in map matching unfortunately. Hope to iron them out soon.

**Comment by kevinkreiser** [comment]
yeah this is a known bug. when doing the interpolated matches we forgot to set the edge index (to the edge that it interpolated to). the fix is straight forward i just havent had time to do it. the edge_index will be omitted if the point doesnt match to anything though

**Comment by M393** [comment]
Ok, thanks for the explanation.

**Comment by sa21pa** [comment]
I have seen that work is already going on to solve the problem. Is it possible to estimate when the solution will be implemented?

**Comment by nilsnolde** [comment]
should hopefully happen next week sometime. @kevinkreiser and me just need to find time to go over the proposed changes (and possibly find a nicer way to do it), which is easier said than done :sweat_smile: gsoc & gtfs takes most of our prio right now.

**Comment by sa21pa** [comment]
that sounds good : ) thank you for your efforts

**Comment by helloln** [comment]
Hi @nilsnolde , has this problem been resolved? I also encountered this problem with version [manually_triggered_build_amd64](https://github.com/orgs/gis-ops/packages/container/docker-valhalla%2Fvalhalla/166744174?tag=manually_triggered_build_amd64)

**Comment by nilsnolde** [comment]
pr was merged, but there's still problems. just follow the breadcrumbs and you'll land here: https://github.com/valhalla/valhalla/issues/3371.

also don't use the "manually triggered" stuff, that's mostly for urgent client access or debugging. latest is the way to go.

**Comment by helloln** [comment]
@nilsnolde I saw this sentence 'stable release Tags (e.g. 3.0.9)' which means 3.0.9 is more stable. I am confused whether I should install latest or 3.0.9. What improvements have been made in latest version 3.0.9? Which version do you recommend I use?

**Comment by nilsnolde** [comment]
> this sentence 'stable release Tags (e.g. 3.0.9)' which means 3.0.9 is more stable

no, it doesn't mean that. language barrier I guess. "e.g." means "for example".

today I actually wouldn't even describe our releases as "more stable", maybe "more rounded". we don't really have a release process as such, we just try to make sense in what's being merged before. so really, releases are as stable or unstable as current master. lots of people just don't like installing things from master, but in our case it hardly matters. however, once we'd move towards package managers again, we'd ideally have more regular releases.

tldr: just go with latest as I said initially. whether or not some bug is fixed for _your_ particular request(s) you'll need to find out yourself. it really just takes 5 minutes.
